src/wrapped_radix_engine_toolkit/high_level/manifest_building/manifests.rs

```diff
@@ -0,0 +1,290 @@
+        value: impl ScryptoToMetadataEntry,
+    ) -> Self
+    where
+        A: IntoScryptoAddress,
+    {
+        let scrypto_manifest = ScryptoManifestBuilder::new()
+            .set_metadata(address.scrypto(), key, value)
+            .build();
+
+        TransactionManifest::from_scrypto(
+            scrypto_manifest,
+            address.network_id(),
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use pretty_assertions::{assert_eq, assert_ne};
+    use rand::Rng;
+
+    #[allow(clippy::upper_case_acronyms)]
+    type SUT = TransactionManifest;
+
+    #[test]
+    fn manifest_for_faucet() {
+        manifest_eq(
+            SUT::faucet(true, &AccountAddress::sample_mainnet()),
+            r#"
+            CALL_METHOD
+                Address("component_rdx1cptxxxxxxxxxfaucetxxxxxxxxx000527798379xxxxxxxxxfaucet")
+                "lock_fee"
+                Decimal("5000")
+            ;
+            CALL_METHOD
+                Address("component_rdx1cptxxxxxxxxxfaucetxxxxxxxxx000527798379xxxxxxxxxfaucet")
+                "free"
+            ;
+            CALL_METHOD
+                Address("account_rdx16xlfcpp0vf7e3gqnswv8j9k58n6rjccu58vvspmdva22kf3aplease")
+                "try_deposit_batch_or_abort"
+                Expression("ENTIRE_WORKTOP")
+                Enum<0u8>()
+            ;
+            "#,
+        );
+    }
+
+    #[test]
+    fn manifest_for_set_account_to_dapp_definition_address() {
+        manifest_eq(
+            SUT::marking_account_as_dapp_definition_type(
+                &AccountAddress::sample_mainnet(),
+            ),
+            r#"
+            SET_METADATA
+                Address("account_rdx16xlfcpp0vf7e3gqnswv8j9k58n6rjccu58vvspmdva22kf3aplease")
+                "account_type"
+                Enum<0u8>(
+                    "dapp definition"
+                )
+            ;
+            "#,
+        );
+    }
+
+    #[test]
+    fn manifest_for_owner_keys() {
+        manifest_eq(
+            SUT::set_owner_keys_hashes(
+                &AccountAddress::sample_mainnet().into(),
+                vec![
+                    PublicKeyHash::hash(Ed25519PublicKey::sample_alice()),
+                    PublicKeyHash::hash(Secp256k1PublicKey::sample_bob()),
+                ],
+            ),
+            r#"
+            SET_METADATA
+                Address("account_rdx16xlfcpp0vf7e3gqnswv8j9k58n6rjccu58vvspmdva22kf3aplease")
+                "owner_keys"
+                Enum<143u8>(
+                    Array<Enum>(
+                        Enum<1u8>(
+                            Bytes("f4e18c034e069baee91ada4764fdfcf2438b8f976861df00557d4cc9e7")
+                        ),
+                        Enum<0u8>(
+                            Bytes("169b4cc19da76c93d4ec3d13ad12cdd5762a8318a643d50f09d0121d94")
+                        )
+                    )
+                )
+            ;
+            "#,
+        );
+    }
+
+    #[test]
+    fn stake_claims() {
```

Comment:
@iamyulong @0xOmarA here is the produced manifest for stake claims, which uses the:

```rust
let bucket = &bucket_factory.next();
```